File: 30 Days of Code/Day 25: Running Time and Complexity.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import math

def is_prime(n):
    if n == 1:
        print("Not prime")
        return
    if n == 2:
        print("Prime")
        return

    for i in range(2, math.ceil(math.sqrt(n)) + 1):
        if n % i == 0:
            print("Not prime")
            return

    print("Prime")


t = int(input())
for _ in range(t):
    is_prime(int(input()))

File: 30 Days of Code/Day 25: Running Time and Complexity.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

void isPrime(int n) {
    bool primeFlag = true;
    if (n == 1)
        primeFlag = false;
    else if (n == 2)
        primeFlag = true;
    else {
        for (int i = 2; i <= (int)ceil(sqrt(n)); i++)
            if (n % i == 0)
                primeFlag = false;
    }

    if (primeFlag)
        cout << "Prime" << endl;
    else
        cout << "Not prime" << endl;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int t;
    cin >> t;
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        isPrime(n);
    }

    return 0;
}
